controllers/internal/controller/helpers/cloud/secrets.go

```diff
@@ -10,30 +10,35 @@ import (
 	"net/http"
 	"time"
 
+	k8sv1 "github.com/Humalect/humalect-core/api/v1"
 	constants "github.com/Humalect/humalect-core/internal/controller/constants"
 	"github.com/aws/aws-sdk-go-v2/aws"
 	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials"
 	"github.com/aws/aws-sdk-go-v2/service/secretsmanager"
 )
 
-func GetCloudSecretMap(azureVaultToken string, secretName string, region string, vaultName string, cloudId string) (map[string]string, error) {
+func GetCloudSecretMap(application *k8sv1.Application) (map[string]string, error) {
 	var secretsMap map[string]string
 	var secretString string
 	var err error
-	if cloudId == constants.CloudIdAWS {
-		secretString, err = getAwsSecretString(secretName, region)
+	if application.Spec.SecretsProvider == constants.CloudIdAWS || (application.Spec.SecretsProvider == "" && application.Spec.CloudProvider == constants.CloudIdAWS) {
+		config, err := config.LoadDefaultConfig(context.TODO(), config.WithCredentialsProvider(credentials.NewStaticCredentialsProvider(application.Spec.AwsSecretCredentials.AccessKey, application.Spec.AwsSecretCredentials.SecretKey, "")))
 		if err != nil {
-			log.Fatal(err.Error())
+			log.Fatal(err)
 			return map[string]string{}, err
 		}
-	} else if cloudId == constants.CloudIdAzure {
-		secretString, err = getAzureSecretString(azureVaultToken, vaultName, secretName)
+
+		secretString, err = getAwsSecretString(config, application.Spec.SecretManagerName, application.Spec.AwsSecretCredentials.Region)
+
+	} else if application.Spec.SecretsProvider == constants.CloudIdAzure || (application.Spec.SecretsProvider == "" && application.Spec.CloudProvider == constants.CloudIdAzure) {
+		secretString, err = getAzureSecretString(application.Spec.AzureVaultCredentials.Token, application.Spec.AzureVaultCredentials.Name, application.Spec.SecretManagerName)
 		if err != nil {
```

Comment:
This is incomplete

Reply:
In `services/aws/get-secret-value.go` we are using session while here, we are using config. for secret fetching. Let's keep both of them consistent.

Reply:
Make it compatible such that if the cluster if AWS and the secretProviderCredentials is not provided then it should work properly as node group has access to secrets